skimage/filters/thresholding.py

```diff
@@ -914,16 +914,23 @@ def threshold_mean(image):
     return np.mean(image)
 
 
-def threshold_triangle(image, nbins=256):
+def threshold_triangle(image=None, nbins=256, *, hist=None):
     """Return threshold value based on the triangle algorithm.
 
+    Either image or hist must be provided. In case hist is given, the actual
+    histogram of the image is ignored.
+
     Parameters
     ----------
     image : (M, N[, ...]) ndarray
         Grayscale input image.
     nbins : int, optional
         Number of bins used to calculate histogram. This value is ignored for
         integer arrays.
+    hist : array, or 2-tuple of arrays, optional
+        Histogram from which to determine the threshold, and optionally a
+        corresponding array of bin center intensities. If no hist provided,
+        this function will compute it from the image.
```

Comment:
You did the right thing in copying a docstring from elsewhere, but I think there's serious room to improve this. As it is it seems unnecessary confusing to me. What do you think about

```suggestion
        An array containing a histogram from which to determine the threshold.
        If not given, the histogram is computed from `image` which is mandatory
        in that case. Optionally, two arrays can be passed to this argument:
        a histogram first and a corresponding array of bin center intensities.
```

And if you want we can improve the other docstrings in a follow-up PR. :)